sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -222,6 +239,23 @@ class StateStoreChangelogWriterV2(
     size += 1
   }
 
+  override def merge(key: Array[Byte], value: Array[Byte]): Unit = {
+    throw new UnsupportedOperationException("Operation not supported with state " +
+      "changelog writer v2")
+  }
+
+  override def merge(key: Array[Byte], value: Array[Byte], colFamilyName: String): Unit = {
+    assert(compressedStream != null)
+    compressedStream.write(RecordType.getRecordTypeAsByte(RecordType.MERGE_RECORD))
```

Comment:
This is same as `PUT` right basically ? can we combine some of the code ?

Reply:
yeah, good point. Done.